components/Map/Map.jsx

```diff
@@ -229,7 +287,14 @@ class Map extends React.Component {
               },
             });
             this.map.once('zoomend', () => {
-              this.setState({ filterGeo: geo });
+              this.setState({
+                filterGeo: geo,
+                minZoom: this.map.getZoom(),
+              });
+              this.map.setMinZoom(this.state.minZoom);
```

Comment:
careful, you are not guaranteed to have the latest version of `this.state.minZoom` here. Either:

```js
this.setState({
  filterGeo: geo,
  minZoom: this.map.getZoom()
});
this.map.setMinZoom(this.map.getZoom())
```

or,
```js
this.setState((prevState) => {
  const newMinZoom = this.map.getZoom();
  this.map.setMinZoom(newMinZoom);
  return {
    filterGeo: geo,
    minZoom: newMinZoom,
})
```

I think I prefer the second option, since it guarantees you are setting minZoom to the map's zoom level *exactly at the time of setting the state*. But either way works, as long as you test your work and it still works as expected.


[source](https://medium.com/@minduladilthushan/resolving-the-state-not-updating-immediately-issue-in-react-js-febf5959c0cf#:~:text=When%20you%20update%20the%20state,return%20the%20old%20state%20value.)

Reply:
I think I agree with the second option as well. I will implement this one!